src/lib/pdfconverter.cc

```diff
@@ -546,13 +546,26 @@ void PdfConverterPrivate::findLinks(QWebFrame * frame, QVector<QPair<QWebElement
 		uexternal  = PageObject::webPageToObject[frame->page()]->settings.useExternalLinks;
 	}
 	if (!ulocal && !uexternal) return;
-	foreach (const QWebElement & elm, frame->findAllElements("a")) {
+	foreach (QWebElement elm, frame->findAllElements("a")) {
 		QString n=elm.attribute("name");
 		if (n.isEmpty()) n=elm.attribute("ns0:name");
 		if (n.startsWith("__WKANCHOR_")) anchors[n] = elm;
 
 		QString h=elm.attribute("href");
-		if (h.isEmpty()) h=elm.attribute("ns0:href");
+		if (h.isEmpty()) {
+			h=elm.attribute("ns0:href");
+
+			//Internal links (anchor tags) are not linked correctly
+			//when the anchor's content is empty. This adds a space with
+			//a height of 0 when it is empty so that internal links
+			//are linked correctly.
+			if(elm.firstChild().isNull()){
```

Comment:
The logic is that if the ```a``` tag has an empty ```href``` attribute then it must be an anchor tag, but it only adds the space if there aren't any children. I tested the following scenarios which all worked:

1. ```<a name="abc"></a>```
2. ```<a id="abc"></a>```
3. ```<a name="abc" href=""></a>```
4. ```<a id="abc" href=""></a>```

If I move the code outside then it will execute for all ```a``` tags and will replace the link text with a space. I tested that just to make sure.